torch/distributed/_functional_collectives.py

```diff
@@ -165,6 +168,30 @@ def _all_gather_into_tensor(shard, tag, ranks, group_size):
 
     return out_tensor
 
+def _reduce_scatter_tensor(
+    input: torch.Tensor,
+    reduceOp: str,
+    scatter_dim: int,
+    tag: str,
+    ranks: List[int],
+    group_size: int,
+):
+    # TODO add dim support?
```

Comment:
I've being thinking about dim support and for it to work we'd need to change the strides of the input and rely on the compiler to propagate that back all the way to the the DTensor local_shard.

It's a nice project on itself, but one we can do later.

Reply:
yeah changing stride requires a copy of the tensor as nccl needs contiguous buffer, we need to think about how to deal with it in the reduce_scatter tensor API, otherwise DTensor can't use (it currently uses the list option as it allows catting back on a different dimension)